store/store.go

```diff
@@ -0,0 +1,127 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/model"
+)
+
+var (
+	// ErrNotFound is returned if a Store cannot find the Alert.
+	ErrNotFound = errors.New("alert not found")
+)
+
+// Alerts implements Store using an in-memory map.
+type Alerts struct {
+	sync.Mutex
```

Comment:
How about a read and write mutex instead so we can have concurrent reads? (https://golang.org/pkg/sync/#RWMutex)